### DISCUSSION TITLE
[web] S3 Image 이름 규칙 정하기

### DISCUSSION THREAD

**Discussion by rkskekzzz** [80000Coding/80000Coding-Web-Client#53] [General]
# S3 Image 이름 규칙 정하기
현재 S3에 올라가는 이미지는 크게 두가지로 나눌 수 있습니다.
* User Profile image
* Content 내부 image

S3에서`test.png` 이름을 업로드하면 아래의 주소로 해당 이미지에 접근할 수 있습니다.
```shell
# S3
https://80000coding.s3.ap-northeast-2.amazonaws.com/{path}/test.png
# Cloudfront
https://d32jhvtzibxeq1.cloudfront.net/{path}/test.png
```

### User profile image 이름 규칙
User profile image는 유저가 각각 하나씩 가지기 때문에 
간단하게 아래와 같은 형태로 하면 좋을 것 같습니다

```
S3 URL + /profile/user_id.png(or jpg, svg ...)
```

### Content 내부 image
content 내부 image는 하나의 content에 여러개의 image가 등록될 수 있습니다. 
따라서 중복되지 않는 이름을 가지는 고유의 규칙이 필요해보입니다. 우선 생각해본 방법은 아래와 같습니다.

```
S3 URL + /content/content_id/timestamp.png(or jpg, svg ...)
```

조금 더 간결한 방법이 있으면 좋을 것 같아서 올려봅니다!! 많은 의견 바랍니다 :)

**Comment by codero225** [comment]
내부 image가 많은데 각각의 timestamp가 어떻게 다르게 되나요?
파일 이름을 contentd_id + 1 부터 순차적으로 주면 어떨까요?

**Reply by rkskekzzz** [reply]
index는 조금 위험하다고 생각이 돼서 다른 방법이 있으면 좋을 것 같습니다 ㅜㅜ
저도 timestamp는 중복이 생길 확률이 있을 것 같아서 고민이네요!

**Reply by codero225** [reply]
index로 지정하면 예상되는 위험한 이유가 뭐가 있나요?

**Reply by rkskekzzz** [reply]
@codero225 규칙이 있다보니, 외부에서 이미지 크롤링 등에 쉽게 노출될 수 있겠다라는 생각을 해봤는데, 해당 문제가 크게 거슬리지 않는 부분이라면 index 형태로 진행해도 될 것 같습니다 ㅎㅎ
